library/src/main/java/com/google/android/exoplayer2/text/eia608/Eia608CueBuilder.java

```diff
@@ -0,0 +1,200 @@
+package com.google.android.exoplayer2.text.eia608;
+
+import com.google.android.exoplayer2.text.Cue;
+import com.google.android.exoplayer2.util.Assertions;
+
+import android.text.Layout;
+import android.text.SpannableStringBuilder;
+import android.text.Spanned;
+import android.text.style.CharacterStyle;
+import android.text.style.ForegroundColorSpan;
+import android.text.style.StyleSpan;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * A Builder for EIA-608 cues.
+ */
+/* package */ final class Eia608CueBuilder {
+
+  private static final int BASE_ROW = 15;
+
+  /**
+   * The caption string.
+   */
+  private SpannableStringBuilder captionStringBuilder;
+
+  /**
+   * The caption styles to apply to the caption string.
+   */
+  private HashMap<Integer, CharacterStyle> captionStyles;
+
+  /**
+   * The row on which the Cue should be displayed.
+   */
+  private int row;
+
+  /**
+   * The indent of the cue - horizontal positioning.
+   */
+  private int indent;
+
+  /**
+   * The setTabOffset offset for the cue.
+   */
+  private int tabOffset;
+
+  public Eia608CueBuilder() {
+    row = BASE_ROW;
+    indent = 0;
+    tabOffset = 0;
+    captionStringBuilder = new SpannableStringBuilder();
+    captionStyles = new HashMap<>();
+  }
+
+  /**
+   * Sets the row for this cue.
+   * @param row the row to set.
+   */
+  public void setRow(int row) {
+    Assertions.checkArgument(row >= 1 && row <= 15);
+    this.row = row;
+  }
+
+  public int getRow() {
+    return row;
+  }
+
+  /**
+   * Rolls up the Cue one row.
+   * @return true if rolling was possible.
+   */
+  public boolean rollUp() {
+    if (row < 1) {
```

Comment:
It seems the condition here should be `if (row <= 1)`. Otherwise when `row` equals `1`, `setRow(0)` will be called and there will be an exception.